### PR TITLE
Add jni method about omp and others

### PR DESCRIPTION
**Pull request by i8run**

### CODE REVIEW

File: mkl2017-xeon-blas/jni/src/main/java/com/intel/analytics/bigdl/mkl/MKL.java [hunk 1]
```diff
@@ -58,8 +58,38 @@ public static String getTmpSoFilePath() {
             return tmpFile.getAbsolutePath();
     }
 
+    // {{ mkl environments set up
+
+    /**
+     * If mkl is linked with parallel mode, one can modify the number of threads omp will use.
+     * It's an subsitute of environment variable: OIMP_NUM_THREADS
```
Comment (yiheng) [inline mkl2017-xeon-blas/jni/src/main/java/com/intel/analytics/bigdl/mkl/MKL.java]: OIMP?

File: mkl2017-xeon-blas/mkl/pom.xml [hunk 2]
```diff
@@ -48,8 +48,10 @@
                         <linkerStartOption>-I ${JAVA_HOME}/include/</linkerStartOption>
                         <linkerStartOption>${linkerStartOptionDirectory}</linkerStartOption>
                         <linkerStartOption>-ldl</linkerStartOption>
+                        <linkerStartOption>-liomp5</linkerStartOption>
                         <linkerStartOption>-shared</linkerStartOption>
-                        <linkerStartOption>-mkl=sequential</linkerStartOption>
+                        <linkerStartOption>-mkl=parallel</linkerStartOption>
+                        <linkerStartOption>-qopenmp</linkerStartOption>
```
Comment (yiheng) [inline mkl2017-xeon-blas/mkl/pom.xml]: what's this for?

File: mkl2017-xeon-dnn/mkl/pom.xml [hunk 3]
```diff
@@ -73,6 +73,7 @@
                         <linkerStartOption>-mkl=parallel</linkerStartOption>
                         <linkerStartOption>-static-intel</linkerStartOption>
                         <linkerStartOption>-qopenmp</linkerStartOption>
+                        <linkerStartOption>-no-intel-extensions</linkerStartOption>
```
Comment (yiheng) [inline mkl2017-xeon-dnn/mkl/pom.xml]: what's this for?

Comment (i8run) [inline mkl2017-xeon-dnn/mkl/pom.xml]: fix: remove warnning  #10237: -lcilkrts linked in dynamically, static library not available

File: mkl2017-xeon-blas/jni/src/main/java/com/intel/analytics/bigdl/mkl/MKL.java (hunk 1 above)
Comment (i8run) [inline mkl2017-xeon-blas/jni/src/main/java/com/intel/analytics/bigdl/mkl/MKL.java]: Typo. OMP_NUM_THREADS

File: mkl2017-xeon-blas/mkl/pom.xml (hunk 2 above)
Comment (i8run) [inline mkl2017-xeon-blas/mkl/pom.xml]: If we want to link mkl with parallel mode, we should change sequential back.

Comment (yiheng) [inline mkl2017-xeon-blas/mkl/pom.xml]: I mean -qopenmp

**Comment by yiheng** [comment]
LGTM

File: mkl2017-xeon-blas/mkl/pom.xml (hunk 2 above)
Comment (i8run) [inline mkl2017-xeon-blas/mkl/pom.xml]: parallel mode must need openmp, it's just an option that manual said.

**Pull request merged**